src/go/k8s/controllers/redpanda/redpanda_controller.go

```diff
@@ -98,14 +93,8 @@ type RedpandaReconciler struct {
 // SetupWithManager sets up the controller with the Manager.
 func (r *RedpandaReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&v1alpha1.Redpanda{}, builder.WithPredicates(
-			predicate.Or(predicate.GenerationChangedPredicate{},
-				predicates.ReconcileRequestedPredicate{}),
-		)).
-		Watches(
-			&source.Kind{Type: &helmv2beta1.HelmRelease{}},
-			&handler.EnqueueRequestForObject{},
-		).
+		For(&v1alpha1.Redpanda{}).
+		Owns(&helmv2beta1.HelmRelease{}).
```

Comment:
I'm not questioning this change, but what predicate was missing, so you changed it to `For` and `Owns`?

Reply:
The previous predicates did not detect changes to annotations, and i believe that was on purpose, here we do watch for them. I decided to keep things simple and use what works.